tests/sentry/tasks/deletion/test_scheduled.py

```diff
@@ -1,111 +1,138 @@
+import abc
+from abc import abstractmethod
 from datetime import timedelta
+from typing import Type
 from unittest.mock import Mock
 
+from django.db.models import QuerySet
+
 from sentry.constants import ObjectStatus
-from sentry.models import Repository, ScheduledDeletion, Team
+from sentry.models import (
+    ApiApplication,
+    ApiApplicationStatus,
+    BaseScheduledDeletion,
+    Repository,
+    Team,
+    get_regional_scheduled_deletion,
+)
+from sentry.services.hybrid_cloud.user import user_service
 from sentry.signals import pending_delete
+from sentry.silo import SiloMode
 from sentry.tasks.deletion.scheduled import reattempt_deletions, run_scheduled_deletions
 from sentry.testutils import TestCase
+from sentry.testutils.silo import control_silo_test, region_silo_test
+
+
+class RegionalRunScheduleDeletionTest(abc.ABC):
```

Comment:
Leveraged existing tests with a base class to enable parameterization and testing all silo modes.  It also validates that monolith mode processes both silo model types.